src/generic_core/core.ts

```diff
@@ -279,7 +280,8 @@ class uProxyCore implements uProxy.CoreAPI {
    * instances.
    */
 
-  public updateGlobalSettings = (newSettings:Core.GlobalSettings) => {
+  public updateGlobalSettings = (newSettings :Core.GlobalSettings) => {
+    newSettings.version = uProxy.STORAGE_VERSION;
```

Comment:
This should already be set to uProxy.STORAGE_VERSION, in general the app and the extension versions should always be the same. Do we have checks for it? This is not really within a scope of this pull request.

For now I'd say throw and error if newSettings.version !== uProxy.STORAGE_VERSION, but again ui doesn't know anything about the storage, so we don't even care.